src/IssueViz.UnitTests/IssueVisualizationControl/IssueVisualizationViewModelTests.cs

```diff
@@ -101,26 +103,109 @@ public void Ctor_InitializeWithExistingSelection()
+        }
+
+        [TestMethod]
+        public void CurrentIssuePropertyChanged_Span_RaisesNotifyPropertyChanged()
+        {
+            var issueViz = CreateIssue();
+
+            RaiseSelectionChangedEvent(SelectionChangeLevel.Issue, issueViz);
+            propertyChangedEventHandler.Reset();
+
+            RaisePropertyChangedEvent(Mock.Get(issueViz), nameof(IAnalysisIssueVisualization.Span));
+
+            VerifyNotifyPropertyChanged(nameof(IIssueVisualizationViewModel.LineNumber), nameof(IIssueVisualizationViewModel.HasNonNavigableLocations));
+        }
+
+        [TestMethod]
+        public void CurrentIssuePropertyChanged_UnknownProperty_NotifyPropertyChangedNotRaised()
+        {
+            var issueViz = CreateIssue();
+
+            RaiseSelectionChangedEvent(SelectionChangeLevel.Issue, issueViz);
+            propertyChangedEventHandler.Reset();
+
+            RaisePropertyChangedEvent(Mock.Get(issueViz), "some unknown property");
+
+            VerifyNotifyPropertyChangedNotRaised();
+        }
+
+        #endregion
+
+        #region Issue Locations PropertyChanged
+
+        [TestMethod]
+        public void CurrentIssueLocationsPropertyChanged_SpanProperty_RaisesNotifyPropertyChanged()
+        {
+            var location = CreateLocation();
+            var flow = CreateFlow(location);
+            var issue = CreateIssue(flows: new[] {flow});
+
+            RaiseSelectionChangedEvent(SelectionChangeLevel.Issue, issue, flow, location);
+
+            propertyChangedEventHandler.Reset();
+
+            RaisePropertyChangedEvent(Mock.Get(location), nameof(IAnalysisIssueLocationVisualization.Span));
+
+            VerifyNotifyPropertyChanged(nameof(IIssueVisualizationViewModel.HasNonNavigableLocations));
+        }
+
+        [TestMethod]
+        public void CurrentIssueLocationsPropertyChanged_UknownProperty_NotifyPropertyChangedNotRaised()
+        {
+            var location = CreateLocation();
+            var flow = CreateFlow(location);
+            var issue = CreateIssue(flows: new[] {flow});
+
+            RaiseSelectionChangedEvent(SelectionChangeLevel.Issue, issue, flow, location);
+            propertyChangedEventHandler.Reset();
+
+            RaisePropertyChangedEvent(Mock.Get(location), "some unknown property");
+
+            VerifyNotifyPropertyChangedNotRaised();
+        }
+
+        #endregion
+
+        #region HasNonNavigableLocations
+
+        [TestMethod]
+        public void HasNonNavigableLocations_CurrentIssueIsNull_False()
+        {
+            testSubject.HasNonNavigableLocations.Should().BeFalse();
+        }
+
+        [TestMethod]
+        [DataRow(false, false, false)]
+        [DataRow(false, true, true)]
+        [DataRow(true, false, true)]
+        [DataRow(true, true, true)]
+        public void HasNonNavigableLocations_ReturnsTrueIfPrimaryOrSecondaryHaveNonNavigableSpans(bool issueHasEmptySpan, bool locationHasEmptySpan, bool expectedNonNavigableLocations)
+        {
```

Comment:
Minor: I think calling the parameters `issueIsNavigable` and `locationIsNavigable` would be clearer.